### PR TITLE
[terra-form-select] Fix onBlur event handling when selecting an option

### LINKED ISSUE #2845: [terra-form-select] Invalid onBlur events triggered when selecting an option
# Bug Report

## Description

The Select component is triggering inappropriate onBlur events when selecting options.

Switching the variant produces different onBlur results.

Default Variant:

A default variant will fire onBlur every time an option is selected. (Mobile and Desktop)

Multiple/Search/Tag/Combobox:

The onBlur fires when the first option is selected and the dropdown menu closes. (Mobile)

Example Code:

```jsx
import React, { useState } from 'react';
import Select from 'terra-form-select';

const Example = () => {
  const [value, setValue] = useState(['blue']);
  const [count, setCount] = useState(0);

  return (
    <div>
      <p>{`Blur Count: ${count}`}</p>
      <Select variant="multiple" onBlur={() => { setCount(count + 1); }} value={value} onChange={setValue}>
        <Select.Option value="blue" display="Blue" />
        <Select.Option value="green" display="Green" />
        <Select.Option value="purple" display="Purple" />
        <Select.Option value="red" display="Red" />
        <Select.Option value="violet" display="Violet" />
      </Select>
    </div>
  );
};

export default Example;
```

## Steps to Reproduce

1. Create a demo page using the code snippet above.
2. Click on the arrow button of the Select to open the dropdown.
3. Select an option.
4. Observe the onBlur event count. 
5. Repeat steps 2-4 with each variant and observe they behave differently.

Note: The behavior is different when the dropdown is opened via clicking the arrow button vs clicking into the text/selection region.

Note: The behavior is different on desktop and mobile. Each variant should be testing on both desktop and mobile.

### Additional Notes

The mobile workflows were tested on mobile iOS Safari using an iPhone X.

## Expected Behavior

The onBlur event should only fire when focus navigates away from the Select. For the tag/multiple variants the dropdown menu should not close when selecting the first option.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
I have added condition, such that whenever `relatedTarget === this.input` onBlur event should not be triggered for mobile.

Resolves #2845

### CODE REVIEW

File: packages/terra-form-select/src/search/Frame.jsx [hunk 1]
```diff
@@ -442,7 +442,7 @@ class Frame extends React.Component {
       searchValue: '',
       hasSearchChanged: false,
       isOpen: false,
-      isAbove: false,
+      // isAbove: false,
```
Comment (ShettyAkarsh) [inline packages/terra-form-select/src/search/Frame.jsx]: you could remove this line. if not required.

File: packages/terra-form-select/src/combobox/Frame.jsx [hunk 2]
```diff
@@ -468,6 +468,8 @@ class Frame extends React.Component {
       isAbove: false,
     });
 
+    this.input.focus();
```
Comment (ShettyAkarsh) [inline packages/terra-form-select/src/combobox/Frame.jsx]: why do we need this to be set explicitly?

File: packages/terra-form-select/src/search/Frame.jsx (hunk 1 above)
Comment (pranav300) [reply packages/terra-form-select/src/search/Frame.jsx]: updated here https://github.com/cerner/terra-core/pull/2904/commits/2d8de879a0187eb87b5dea6d6877d871738c1ef4

File: packages/terra-form-select/src/combobox/Frame.jsx (hunk 2 above)
Comment (pranav300) [reply packages/terra-form-select/src/combobox/Frame.jsx]: Removed here https://github.com/cerner/terra-core/pull/2904/commits/2d8de879a0187eb87b5dea6d6877d871738c1ef4

File: packages/terra-form-select/src/combobox/Frame.jsx [hunk 3]
```diff
@@ -329,7 +329,7 @@ class Frame extends React.Component {
     // and if the relatedTarget is falsey. The relatedTarget will be null when
     // dismissing the onscreen keyboard, else set to another element when
     // tapping elsewhere on the page
-    if (focusedByTouch && !relatedTarget) {
+    if (focusedByTouch && (relatedTarget === this.input)) {
```
Comment (nramamurth) [inline packages/terra-form-select/src/combobox/Frame.jsx]: This inline comment ⬇️ may need to be updated now.

> Determined by if we have have interacted with the frame via onTouchStart and if the relatedTarget is falsey.

File: packages/terra-form-select/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,8 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
+* Fixed `onBlur` event to not be triggered when selecting an option
```
Comment (nramamurth) [inline packages/terra-form-select/CHANGELOG.md]: Does this need to be a bit more specific that this was done _just_ for mobile?

**Comment by StephenEsser** [comment]
With these changes desktop is still experiencing invalid onBlur/onFocus. Are you wanting to split the changes into two separate PRs to fix desktop/mobile issues?

File: packages/terra-form-select/src/combobox/Frame.jsx (hunk 3 above)
Comment (pranav300) [reply packages/terra-form-select/src/combobox/Frame.jsx]: Updated here https://github.com/cerner/terra-core/pull/2904/commits/09eb6be42fd6ff175b3d324f8d5ccbb9ca1583ec

File: packages/terra-form-select/CHANGELOG.md (hunk 4 above)
Comment (pranav300) [reply packages/terra-form-select/CHANGELOG.md]: Updated here https://github.com/cerner/terra-core/pull/2904/commits/09eb6be42fd6ff175b3d324f8d5ccbb9ca1583ec

**Comment by StephenEsser** [comment]
One of the WDIO screenshots for the select is failing. Do we know what caused the change and if the screenshot should be updated?

cc: @pranav300

**Pull request merged**